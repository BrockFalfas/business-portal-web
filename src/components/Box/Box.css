.Box {
  background-color: #fff;
  border-radius: 15px;
  box-shadow: 0 0 6px 2px rgba(0, 0, 0, 0.08);
}

.Box .ant-table-thead > tr:first-child > th:first-child {
  border-top-left-radius: 15px;
}
.Box .ant-table-thead > tr:first-child > th:last-child {
  border-top-right-radius: 15px;
}
.Box .ant-table-body > tr:last-child > th:first-child {
  border-bottom-left-radius: 15px;
}
.Box .ant-table-body > tr:last-child > th:last-child {
  border-bottom-right-radius: 15px;
}
.Box .ant-pagination {
  margin-right: 1rem;
}
