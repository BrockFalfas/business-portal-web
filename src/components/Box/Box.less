@import "../../styles/variables.less";

@border-radius: 0.9rem;

.Box {
  background-color: #fff;
  border-radius: @border-radius;
  box-shadow: @box-shadow;
  margin-bottom: 1rem;

  &--transparent {
    background-color: transparent;
    border-radius: 0;
    box-shadow: none;
  }

  /* antd <Table /> */
  .ant-table-thead > tr:first-child > th:first-child {
    border-top-left-radius: @border-radius;
  }
  .ant-table-thead > tr:first-child > th:last-child {
    border-top-right-radius: @border-radius;
  }
  .ant-table-body > tr:last-child > th:first-child {
    border-bottom-left-radius: @border-radius;
  }
  .ant-table-body > tr:last-child > th:last-child {
    border-bottom-right-radius: @border-radius;
  }
  .ant-table-pagination.ant-pagination {
    margin-right: 1rem;
  }
}
