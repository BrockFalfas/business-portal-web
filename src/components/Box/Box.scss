@import "styles/variables";

.Box {
  background-color: #fff;
  border-radius: $border-radius;
  box-shadow: $box-shadow;

  /* antd <Table /> */
  .ant-table-thead > tr:first-child > th:first-child {
    border-top-left-radius: $border-radius;
  }
  .ant-table-thead > tr:first-child > th:last-child {
    border-top-right-radius: $border-radius;
  }
  .ant-table-body > tr:last-child > th:first-child {
    border-bottom-left-radius: $border-radius;
  }
  .ant-table-body > tr:last-child > th:last-child {
    border-bottom-right-radius: $border-radius;
  }
  .ant-table-pagination.ant-pagination {
    margin-right: 1rem;
  }
}
