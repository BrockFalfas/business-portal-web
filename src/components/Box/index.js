import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Box.css';

export default class Box extends React.Component {
  static propTypes = {
    backgroundColor: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    className: PropTypes.string,
    transparent: PropTypes.bool,
    padding: PropTypes.bool,
  };

  static defaultProps = {
    backgroundColor: 'white',
    className: null,
    transparent: false,
    padding: false,
  };

  render() {
    const { children, className, backgroundColor, transparent, padding } = this.props;

    return (
      <div
        className={classnames('Box', {
          [className]: className,
          [`Box--bg-${backgroundColor}`]: true,
          'Box--transparent': transparent,
          'Box--padding': padding,
        })}>
        {children}
      </div>
    );
  }
}
