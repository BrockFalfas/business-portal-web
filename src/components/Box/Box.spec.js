import React from 'react';
import { shallow } from 'enzyme';

import Box from './index';

describe('component: <Box/>', () => {
  describe('Box', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<Box />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render children properly', () => {
      const fakeChild = <div>fake child</div>;
      const wrapper = shallow(<Box>{fakeChild}</Box>);
      expect(wrapper).toMatchSnapshot();
    });

    it('should allow setting custom className', () => {
      const wrapper = shallow(<Box className="fake1 fake2" />);
      expect(wrapper.prop('className')).toContain('fake1 fake2');
    });

    it('should allow setting transparent background', () => {
      const wrapper = shallow(<Box transparent />);
      expect(wrapper.prop('className')).toContain('Box--transparent');
    });

    it('should allow setting padding', () => {
      const wrapper = shallow(<Box padding />);
      expect(wrapper.prop('className')).toContain('Box--padding');
    });
  });
});
