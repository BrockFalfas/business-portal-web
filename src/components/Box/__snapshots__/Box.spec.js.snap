// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Box/> Box should render children properly 1`] = `
<div
  className="Box Box--bg-white"
/>
`;

exports[`component: <Box/> Box should render without crashing 1`] = `
<div
  className="Box Box--bg-white"
/>
`;
